administration/views.py

```diff
@@ -1,25 +1,27 @@
 from rest_framework.permissions import IsAuthenticated, IsAdminUser
-from rest_framework.generics import ListAPIView, RetrieveUpdateDestroyAPIView
+from rest_framework.generics import ListCreateAPIView, RetrieveUpdateDestroyAPIView
+from rest_framework.response import Response
 
-from authentication.models import CustomUser
-from .serializers import AdminUserSerializer
-from .pagination import UserListPagination
+from administration.serializers import AdminCompanySerializer
+from administration.pagination import ListPagination
+from profiles.models import Profile
 
 
-class UsersListView(ListAPIView):
-    """
-    List of users.
-    """
-    queryset = CustomUser.objects.filter(is_superuser=False).order_by('id')
-    serializer_class = AdminUserSerializer
-    permission_classes = [IsAuthenticated, IsAdminUser]
-    pagination_class = UserListPagination
+class AdminProfileList(ListCreateAPIView):
+    permission_classes = (IsAuthenticated, IsAdminUser)
+    pagination_class = ListPagination
+    serializer_class = AdminCompanySerializer
+    queryset = Profile.objects.filter(is_deleted=False)
+    
 
-
-class UserDetailView(RetrieveUpdateDestroyAPIView):
-    """
-    Retrieve, update or delete a user.
-    """
-    queryset = CustomUser.objects.filter(is_superuser=False).order_by('id')
-    permission_classes = [IsAuthenticated, IsAdminUser]
-    serializer_class = AdminUserSerializer
+class AdminProfileDetail(RetrieveUpdateDestroyAPIView):
+    permission_classes = (IsAuthenticated, IsAdminUser)
+    serializer_class = AdminCompanySerializer
+    queryset = Profile.objects.filter(is_deleted=False)
+    
+    def update(self, request, *args, **kwargs):
+        instance = self.get_object()
+        serializer = self.get_serializer(instance, data=request.data, partial=True)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(serializer.data)
```

Comment:
Do we need this method? It looks like it does almost exactly the same what does built-in DRFs implementation. Probably only making all updates 'partial' (and I'd believe it's not a good idea to do this).

Do we have to do `return Response()` here?

Reply:
Remove this method.